save-core/src/jvmTest/kotlin/org/cqfn/save/core/integration/GeneralTest.kt

```diff
@@ -0,0 +1,88 @@
+package org.cqfn.save.core.integration
+
+import org.cqfn.save.core.files.readFile
+import org.cqfn.save.core.result.Pass
+import org.cqfn.save.core.utils.CurrentOs
+import org.cqfn.save.core.utils.ProcessBuilder
+import org.cqfn.save.core.utils.getCurrentOs
+import org.cqfn.save.core.utils.isCurrentOsWindows
+import org.cqfn.save.reporter.Report
+
+import okio.FileSystem
+import okio.Path.Companion.toPath
+
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+@Suppress(
+    "TOO_LONG_FUNCTION",
+    "INLINE_CLASS_CAN_BE_USED",
+    "MISSING_KDOC_TOP_LEVEL",
+    "MISSING_KDOC_CLASS_ELEMENTS")
+@OptIn(ExperimentalSerializationApi::class)
+class GeneralTest {
+    private val fs = FileSystem.SYSTEM
+
+    @Test
+    fun `examples test`() {
+        val binDir = "../save-cli/build/bin/" + when (getCurrentOs()) {
+            CurrentOs.LINUX -> "linuxX64"
+            CurrentOs.MACOS -> "macosX64"
+            CurrentOs.WINDOWS -> "mingwX64"
+            else -> return
+        } + "/debugExecutable"
+
+        assertTrue(fs.exists(binDir.toPath()))
+
+        val saveExecutableFiles = fs.list(binDir.toPath()).filter { fs.metadata(it).isRegularFile }
+        // Binary should be created at this moment
+        assertTrue(saveExecutableFiles.isNotEmpty())
+
+        val examplesDir = "../examples/kotlin-diktat/"
+
+        val actualSaveBinary = saveExecutableFiles.last()
+        val saveBinName = if (isCurrentOsWindows()) "save.exe" else "save"
+        val destination = (examplesDir + saveBinName).toPath()
+
+        // Copy latest version of save into examples
+        fs.copy(actualSaveBinary, destination)
+        assertTrue(fs.exists(destination))
+        fs.delete(actualSaveBinary)
```

Comment:
Why do you delete it?

Reply:
This binary we use just for tests, so if during the development we will usually rerun tests and hold this file I think
`save-cli/bin/..` will contain a lot of garbage

Reply:
But this is the file, produced by `linkDebugExecutable` task, right? I think it will be very confusing, if it gets deleted during tests. Also, during development they all will have the same name with snapshot version.

Reply:
Yes, but this task we call by ourselves in test. So I don't see any confusing: we create bin file for our needs, and then remove it.

I didn't met situations in SAVE-cli, when I get an snapshot version by `linkDebugExecutable` However, it make sence,
if we will get such situation, then we will remove the only one existing binary.

Removed this line